src/Kiota.Builder/PathSegmenters/GoPathSegmenter.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections.Generic;
+using System.Linq;
+using Kiota.Builder.Extensions;
+
+namespace Kiota.Builder {
+    public class GoPathSegmenter : CommonPathSegmenter
+    {
+        public GoPathSegmenter(string rootPath, string clientNamespaceName) : base(rootPath, clientNamespaceName) {}
+        public override string GetFileSuffix(CodeElement currentElement) => ".go";
```

Comment:
With the latest update there is no need for the `IPathSegmenter` to enforce the `CodeElement` parameter as this was the only "implementer" using it. 
Do we see this being needed or should we just clean it up?

Reply:
great catch! reverted.